worker/src/helper.rs

```diff
@@ -1,115 +0,0 @@
-// Copyright (c) 2021, Facebook, Inc. and its affiliates
```

Comment:
> This is great work @aschran 💯 and a pleasure to see us adopting a request/response pattern for such things. I am already thinking of other places where this would simplify things. Left a few comments. Could you also please run the benchmarks (apologies if you have done already) to ensure we don't see any significant change in performance - although I am not 100% sure the [current benchmark](https://github.com/MystenLabs/narwhal/tree/main/benchmark)s will reveal anything as this flow is not a frequently executed one unless we are under some kind of stress (ex network delays etc)

Thanks for the reminder. So it looks like I broke something with the benchmark (or else caused a big performance problem)? Lol. I'm not sure how consensus TPS could drop that much but e2e TPS stay the same? Maybe a log line the parser wants is no longer emitted? I'm also confused about the negative latency reports? Which happen even on main branch, apparently.

I'll start digging.

Before:
```
 + RESULTS:
 Consensus TPS: 48,849 tx/s
 Consensus BPS: 25,010,596 B/s
 Consensus latency: -77,277 ms

 End-to-end TPS: 48,515 tx/s
 End-to-end BPS: 24,839,538 B/s
 End-to-end latency: -65,391 ms
```

After:
```
+ RESULTS:
 Consensus TPS: 38 tx/s
 Consensus BPS: 19,641 B/s
 Consensus latency: -126,861 ms

 End-to-end TPS: 48,922 tx/s
 End-to-end BPS: 25,047,886 B/s
 End-to-end latency: -259,855 ms
```

Reply:
OK, I'm now realizing these results are inconsistent in a way that doesn't seem related to this PR.

I ran it a few more times, and some runs with my changes look fine:
```
 + RESULTS:
 Consensus TPS: 48,696 tx/s
 Consensus BPS: 24,932,343 B/s
 Consensus latency: -51,812 ms

 End-to-end TPS: 48,308 tx/s
 End-to-end BPS: 24,733,859 B/s
 End-to-end latency: -49,343 ms
```

Meanwhile on unmodified `main`, I can reproduce the bad results with similar frequency, with failures due to `BlockWaiter` timeouts and consensus TPS numbers of 38tx/s.

Hopefully continuing on with this refactoring makes exactly that sort of issue easier to debug? Right now it's hard for me to tell the cause and if this change makes it worse or has no effect.